### PR TITLE
Enumerable modules

### PR DESCRIPTION
**Pull request by LOctavio**
In this project, I created methods that do the same work than the original methods from the Enumerable module.

- I created a module to put all the methods inside
- I used the method my_each in all the other methods except the my_each_with_index
- All the methods can take a block

### CODE REVIEW

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+  def my_each_with_index
+    return self unless block_given?
+
+    i = 0
+    while i < length
+      yield self[i], i
+      i += 1
+    end
+  end
```
Comment (stratospherique) [inline enumerable.rb]: * [x] should have working `my_each_with_index` identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [x] my_each_with_index returns an enumerator if no block is given
  ```ruby
    array = [1,2,5,8,7,98,4]
    p array.my_each_with_index.class # not Enumerator
  ```

File: enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+  def my_none?(pattern = nil)
+    val = true
+    if pattern.nil?
+      block_given? ? my_each { |x| val = false if yield x } : val = my_any? { |x| x == true }
+    else
+      my_each { |x| val = false if pattern === x }
+    end
+    val
+  end
```
Comment (stratospherique) [inline enumerable.rb]: * [x] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [x] my_none? when no block or argument is given returns true only if none of the collection members is true
  ```ruby
    array = [nil, false, nil, false]
    p array.my_none? == array.none? # false
  ```

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great work @LOctavio :clap: :clap: Nice start. However, some methods are behaving differently than expected. Also, other changes need to be implemented in order for us to approve your project. Check :arrow_down: 

- **[REQUIRED]** Check the code comments below, they should help you refactor your methods :gear: 

- **[REQUIRED]** Remove all comments or debugging related.
  - You can save the "puts" lines for the testing project.

### That's it :relieved: When finished, kindly resubmit another code review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

File: .rubocop.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  Enabled: true
+
+Style/HashEachMethods:
+  Enabled: true
+
+Style/HashTransformKeys:
+  Enabled: true
+
+Style/HashTransformValues:
+  Enabled: true
```
Comment (shshamim63) [inline .rubocop.yml]: You should not alter the default config file. Check out this [link](https://github.com/microverseinc/linters-config/tree/master/ruby).
image

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## Status :traffic_light: Changes Required :tractor: 
___
Hi @LOctavio :wave: ,
Awesome job :clap:  The project contains a readme,  all enumerable methods are properly working. But found an issue that kept me from approving the project.
___
### :spades: Mandatory Changes :spades: 
___
- You should not alter the basic config template of the linter config. Check the comment below, please.
___
### :clubs: Optional Changes :clubs: 
___
- Well, the previous reviewer mistakenly missed a point. That is related to the `multiply_els` method. So I am marking it as optional. But to complete the oding project instruction, it is needed.
`multiply_els` method. it uses `my_inject` as described on Odin ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/multiply_els.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2)). Remember, it is not an enumerable method, so should be placed outside of the module.
```ruby
def multiply_els(arr)
  arr.my_inject(2, :*)
end
```
___
#### Please fix the issue mentioned above and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](linkedin.com/in/shakhawathossainshamim/)

**Comment by LOctavio** [comment]
I have removed the extra configuration from .rubocom.yml file, but there is an error if I don't use that configuration

**Comment by shshamim63** [comment]
### :1234: Required Changes :1234: 
Are you sure you have added [this](https://github.com/LOctavio/enumerable-methods/blob/add-enumerable-module/.rubocop.yml) config file? Cause all the errors are covered here.

**Comment by LOctavio** [comment]
I hadn't updated the .rubocop.yml file to its new version, but now the changes are done

**Review by mcrd25 — APPROVED** [review]
# Changes Approved 💪 ㊗️ 🚀 

Hi @LOctavio! 👋 

You've successfully made changes previously requested! No need for further review!

***If you have any questions, now or in the future, do not hesitate to contact me on social media, email or on slack and I will be glad to chat with you.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

**Pull request merged**